Src/Web/Web.Shared.Net/Implementation/HttpContextExtensions.cs

```diff
@@ -37,5 +38,24 @@ public static HttpResponse GetResponse(this HttpContext context)
 
             return result;
         }
+
+        public static RequestNotification TryGetCurrentNotification(this HttpContext context)
+        {
+            RequestNotification notification = default(RequestNotification);
+            try
+            {
+                notification = context.CurrentNotification;
+            }
+            catch (NullReferenceException exp)
```

Comment:
Let's add a comment. It is not obvious because this exception is not documented and looks like we try to catch our own exception.